### PR TITLE
docs: simplify import map generation

### PR DESCRIPTION
**Pull request by bennypowers**
Hopefully this is easier to understand and maintain

Depends on #1307

## What I did

1. clean up importMap code, replacing several truly awful hacks with a smaller number of merely gross hacks.

## Testing Instructions

1. pages, demos, playgrounds

### CODE REVIEW

File: docs/_plugins/importMap.cjs [hunk 1]
```diff
@@ -48,56 +58,35 @@ async function getCachedImportMap({
 
       const { Generator } = await import('@jspm/generator');
 
+      const nothing = Symbol();
+      const providers = {
+        '@patternfly': 'nodemodules',
+        ...Object.fromEntries(localPackages?.map(packageName =>
+          packageName.match(/@(rhds|patternfly)/) ? [nothing] : [packageName, 'nodemodules']) ?? []),
+      };
```
Comment (zeroedin) [inline docs/_plugins/importMap.cjs:66]: More of a missing understanding thing.  I was trying to figure out if `nodemodules` was a typo, started looking at docs for jspm generator, couldn't find an explanation for why `nodemodules` was written this way vs `node_modules` as typical.

Is this due to the file name of the provider also being compacted? 
https://github.com/jspm/generator/blob/63ad57d21e5aeb2a6678b9a4a6da455fa75648cc/src/providers/nodemodules.ts

Have I dug to far? 🙃

Comment (zeroedin) [reply docs/_plugins/importMap.cjs:66]: Also not blocking feel free to resolve with no explanation.

Comment (bennypowers) [reply docs/_plugins/importMap.cjs:66]: yeah im' not really sure why, this is just what Guy Bedford called it :shrug:

**Review by zeroedin — COMMENTED** [review]
When loading a playground demos:

```
index.html:1 Uncaught TypeError: Failed to resolve module specifier "@rhds/elements/rh-button/rh-button.js". Relative references must start with either "/", "./", or "../".
```
```
index.html:1 Uncaught TypeError: Failed to resolve module specifier "@rhds/elements/rh-accordion/rh-accordion.js". Relative references must start with either "/", "./", or "../".
```

**Comment by zeroedin** [comment]
> When loading a playground demos:
> 
> ```
> index.html:1 Uncaught TypeError: Failed to resolve module specifier "@rhds/elements/rh-button/rh-button.js". Relative references must start with either "/", "./", or "../".
> ```
> 
> ```
> index.html:1 Uncaught TypeError: Failed to resolve module specifier "@rhds/elements/rh-accordion/rh-accordion.js". Relative references must start with either "/", "./", or "../".
> ```

This seems to be an issue on `main` right now too

**Comment by bennypowers** [comment]
yeah the playgrounds don't work in localhost right now. check the dp

File: eleventy.config.cjs [hunk 2]
```diff
@@ -57,37 +59,44 @@ module.exports = function(eleventyConfig) {
 
   /** Bespoke import map for ux-dot pages and demos */
   eleventyConfig.addPassthroughCopy({ 'node_modules/@lit/reactive-element': '/assets/packages/@lit/reactive-element' });
+  eleventyConfig.addPassthroughCopy({ 'elements': 'assets/packages/@rhds/elements/elements/' });
+  eleventyConfig.addPassthroughCopy({ 'lib': 'assets/packages/@rhds/elements/lib/' });
   eleventyConfig.addPlugin(ImportMapPlugin, {
-    defaultProvider: 'nodemodules',
+    // TODO(bennypowers): There must be a better way
+    baseUrl: isWatch ? 'http://localhost:8080' : 'https://ux.redhat.com',
```
[the remark this answers is not in the document]
Comment (zeroedin) [reply eleventy.config.cjs:66]: after digging a bit more i'm now more confused on where the `baseUrl` gets used after being passed in as an option.  Is the idea to extend on this somewhere thats currently not in code?  I'm missing something

[the remark this answers is not in the document]
Comment (bennypowers) [reply eleventy.config.cjs:66]: maybe we don't need it anymore. feel free to delete

**Review by zeroedin — APPROVED** [review]
Lazy Goats Take Naps

**Pull request merged**